experimental/dds/tree2/src/feature-libraries/editable-tree-2/lazyField.ts

```diff
@@ -157,7 +157,7 @@ export abstract class LazyField<TKind extends FieldKind, TTypes extends AllowedT
 		return this[cursorSymbol].getFieldLength();
 	}
 
-	public at(index: number): UnboxNodeUnion<TTypes> {
+	public atIndex(index: number): UnboxNodeUnion<TTypes> {
```

Comment:
Do we want this to be public?

Reply:
Also, since this is no longer inheriting from the interface, we should document it, especially since it differs from `at` in terms of how `index` is consumed.

Reply:
If we move everything to the new `at` then I don't think `atIndex` is necessary, but that's a bit of my hesitation from the PR description. Not sure if it might make sense to maintain an API with this signature and semantics on LazyField to access its nodes (explicitly constraining how that works in comparison to LazySequence's `at` with `Array.prototype.at` semantics).

Reply:
In the Proxy APIs up above, we're only going to want `at`, because we'll also have the `[index]` operator. Since we can't do `[index]` here, I think it makes sense to have `at` and `atIndex` both. 

Easier to remove than to add if we change our mind later.

Reply:
@noencke `at` ends up disappearing from `LazyField`, it only exists on the `Sequence` interface and `LazySequence` class. `LazyField` ends up using `atIndex` for everything. I'll merge this for now but if you think we need to change that I can put out a separate PR.